pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,271 @@
+    else:
+        rotations = np.eye(3)
+    strain = np.einsum('nik,mkl,njl->nmij', rotations, strain, rotations).reshape(-1, 3, 3)
+    strain = np.stack((strain[:,0,0], strain[:,1,1], strain[:,2,2], 2*strain[:,1,2], 2*strain[:,0,2], 2*strain[:,0,1]), axis=-1)
+    if stress is not None and len(stress)*len(rotations)==len(strain):
+        coeff, score = _fit_coeffs_with_stress(strain=strain, stress=stress, rotations=rotations)
+    elif energy is not None and volume is not None and len(energy)*len(rotations)==len(strain) and len(energy)==len(volume):
+        coeff, score = _fit_coeffs_with_energies(strain=strain, energy=energy, volume=volume, rotations=rotations)
+    else:
+        raise ValueError('Problem with fitting data')
+    if return_score:
+        return np.array(coeff), score
+    else:
+        return np.array(coeff)
+
+def calc_elastic_constants(elastic_tensor):
+    """
+    Calculate elastic constants from the elastic tensor.
+
+    For anistropic material (i.e. zener_ratio!=1), the values may or may not make
+    sense -> don't trust the results straightforwardly
+
+    Args:
+        elastic_tensor (numpy.ndarray): 6x6 tensor
+    """
+    output = {}
+    output['elastic_tensor'] = elastic_tensor
+    output['lame_coefficient'] = np.mean(elastic_tensor[:3, :3].diagonal())
+    output['shear_modulus'] = np.mean(elastic_tensor[3:, 3:].diagonal())
+    output['bulk_modulus'] = np.mean(elastic_tensor[:3,:3])
+    output['youngs_modulus'] = 1/np.mean(np.linalg.inv(elastic_tensor[:3,:3]).diagonal())
+    output['poissons_ratio'] = -output['youngs_modulus']*np.sum(np.linalg.inv(elastic_tensor[:3,:3]))/6+0.5
+    output['zener_ratio'] = 12*np.mean(elastic_tensor[3:,3:].diagonal())/(3*np.trace(elastic_tensor[:3,:3])-np.sum(elastic_tensor[:3,:3]))
+    return output
+
+class ElasticJobGenerator(JobGenerator):
+    @property
+    def parameter_list(self):
+        """
+
+        Returns:
+            (list)
+        """
+        parameter_lst = []
+        for ii, epsilon in enumerate(self._job.input['strain_matrices']):
+            basis = self._job.ref_job.structure.copy()
+            basis.apply_strain(np.array(epsilon))
+            parameter_lst.append([ii, basis])
+        return parameter_lst
+
+    def job_name(self, parameter):
+        return "{}_{}".format(self._job.job_name, parameter[0]).replace(".", "_")
+
+    @staticmethod
+    def modify_job(job, parameter):
+        job.structure = parameter[1]
+        return job
+
+# ToDo: not all abstract methods implemented
+class ElasticTensor(AtomisticParallelMaster):
+    """
+    Class to calculate the elastic tensor and isotropic elastic constants
+
+    Example:
+
+    >>> job = pr.create_job('SomeAtomisticJob', 'atomistic')
+    >>> job.structure = pr.create_structure('Fe', 'bcc', 2.83)
+    >>> elastic = job.create_job('ElasticTensor', 'elastic')
+    >>> elastic.run()
+
+    The input parameters might not be chosen adequately. If you have a large
```

Comment:
If the naming/setup of input parameters is your only concern, I'm happy to merge directly into pyiron even without this sentence.  If turns out we want to tweak them we can always do that before bumping the minor version.

Reply:
The only thing I get afraid of there is someone using this in a notebook and then demanding backwards compatibility. Or someone demanding backwards compatibility whether anyone's using it or not. 

...In case your detecting a theme from my github activity this week, backwards compatibility is something that keeps me awake at night 😆

Reply:
I feel it's reasonable to assume nothing enters public API between version bumps.   Then this class would only become visible after the next version bump, this is what the last part of my previous comment refers to.